apps/meteor/ee/app/livechat-enterprise/server/hooks/afterInquiryQueued.ts

```diff
@@ -1,32 +1,28 @@
+import type { ILivechatInquiryRecord } from '@rocket.chat/core-typings';
 import moment from 'moment';
 
 import { settings } from '../../../../../app/settings/server';
 import { callbacks } from '../../../../../lib/callbacks';
 import { OmnichannelQueueInactivityMonitor } from '../lib/QueueInactivityMonitor';
 import { cbLogger } from '../lib/logger';
 
-let timer = 0;
+const withTimer = (timer: number) => {
```

Comment:
I know you mentioned the task for creating the tests for this in a separated PR, but I think it would be good to have a unit test for this function at least. This way we can ensure the high order function is behaving as expected.

Reply:
Added 😬